src/Microsoft.NetCore.Analyzers/Core/Security/ReviewCodeForSqlInjectionVulnerabilities.cs

```diff
@@ -0,0 +1,133 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.NetCore.Analyzers.Security
+{
+    using System.Linq;
+    using System.Collections.Immutable;
+    using Analyzer.Utilities.Extensions;
+    using Analyzer.Utilities.FlowAnalysis.Analysis.TaintedDataAnalysis;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.Diagnostics;
+    using Microsoft.CodeAnalysis.FlowAnalysis.DataFlow;
+    using Microsoft.CodeAnalysis.Operations;
+    using Microsoft.NetCore.Analyzers.Security.Helpers;
+
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public class ReviewCodeForSqlInjectionVulnerabilities : DiagnosticAnalyzer
+    {
+        internal static readonly DiagnosticDescriptor Rule = SecurityHelpers.CreateDiagnosticDescriptor(
+            "CA3001",
+            nameof(MicrosoftNetCoreSecurityResources.ReviewCodeForSqlInjectionVulnerabilitiesTitle),
+            nameof(MicrosoftNetCoreSecurityResources.ReviewCodeForSqlInjectionVulnerabilitiesMessage),
+            isEnabledByDefault: false,
+            helpLinkUri: null); // TODO paulming: Help link.
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+
+            context.RegisterCompilationStartAction(
+                compilationContext =>
+                {
+                    WellKnownTypeProvider wellKnownTypeProvider = WellKnownTypeProvider.GetOrCreate(compilationContext.Compilation);
+                    TaintedDataSymbolMap<SourceInfo> sourceInfoSymbolMap = new TaintedDataSymbolMap<SourceInfo>(
+                        wellKnownTypeProvider, 
+                        WebInputSources.SourceInfos);
+                    if (sourceInfoSymbolMap.IsEmpty)
```

Comment:
>sourceInfoSymbolMap [](start = 24, length = 19)

I am presuming all the TaintAnalysis based analyzers will have similar logic to create tainted symbol maps for sources and sinks, check if either of them are empty, and if so bail out. Otherwise, continue analysis and pass it down to the core flow analysis.

Consider extracting out the logic and core data structures into a common helper or wrapper type, say `TaintAnalysisInfoProvider` or some such that is analyzer facing, we have something [similar for dispose analysis](https://github.com/dotnet/roslyn-analyzers/blob/master/src/Microsoft.CodeQuality.Analyzers.Exp/Core/DisposeAnalysisHelper.cs#L78), which is used by all dispose rules.

Reply:
Aye, when implementing another tainted data rule, I was thinking of putting most of the logic in base analyzer class, which should cover most cases (Exception might the exception).